CMakeLists.txt

```diff
@@ -520,6 +520,7 @@ endif ()
 # to have priority over any pre-existing/system-wide headers need to appear early in
 # compiler search path.
 include_directories(BEFORE ${broker_includes})
+include_directories(BEFORE ${CMAKE_CURRENT_SOURCE_DIR}/auxil/filesystem/include)
```

Comment:
Just wondering, there's certainly a lot of precedence here to use `include_directories`, but that's really the most global hammer one could use. Since [this library provides a target one could depend on](https://github.com/gulrak/filesystem/blob/cd6805e94dd5d6346be1b75a54cdc27787319dd2/CMakeLists.txt#L39) a better approach might be to just add it to targets which depend on it instead.

Reply:
Thanks for these pointers. I'm not super familiar with cmake (and yes, just copied the approach from the other submodules).

I looked a bit and my high-level understanding is to use `target_link_libraries()`. Outside of plugins, I don't think there are separate targets, and for static/builtin plugins target_link_libraries() doesn't seem to be used and I failed to make it work right.

Though for static/builtin plugins, may also be okay to do the `include_directories` within their local CMakeLists.txt instead of having a global one. Then, for util.h/util.cc (as a candidate using ghc::filesystem), that could be transitioned to a library via `add_library()`.

So for this PR maybe just remove the top-level include_directories and figure out the users when needed? Other thoughts?

Reply:
Actually for consistency I'd just go with the same style as for the other libraries for now. We'll eventually make this all nicer / more modern with a larger CMake overhaul.